pkg/validation/validation.go

```diff
@@ -0,0 +1,51 @@
+package validation
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const ErrorNotFoundAgent = "unsupported fence agent %s:"
+
+type validateAgentExistence struct {
+	isFileExist func(string) (fs.FileInfo, error)
+}
+
+type AgentValidator interface {
+	ValidateAgentName(agent string) (admission.Warnings, error)
+}
+
+func NewAgentValidator(fileGetter func(string) (fs.FileInfo, error)) AgentValidator {
+	return &validateAgentExistence{isFileExist: fileGetter}
+}
+
+func (vfe *validateAgentExistence) ValidateAgentName(agent string) (admission.Warnings, error) {
+	exist, err := vfe.isAgentSupported(agent)
+	if !exist {
+		return admission.Warnings{"Check the agent spec field"}, fmt.Errorf(ErrorNotFoundAgent+" %w", agent, err)
+	}
+	return nil, nil
+
+}
+
+// isAgentSupported returns true if the agent name matches a binary, and false otherwise
+func (vfe *validateAgentExistence) isAgentSupported(agent string) (bool, error) {
+	directory := "/usr/sbin/"
+	// Create the full path by joining the directory and filename
+	fullPath := filepath.Join(directory, agent)
+
+	// Check if the file exists
+	_, err := vfe.isFileExist(fullPath)
+	if err == nil {
+		fmt.Printf("Agent %s was found at %s\n", agent, directory)
```

Comment:
Probably we can drop this `fmt.Printf`, I don't think they are visible in the logs

Reply:
I am pretty confident they do, but it might be better to properly use logs function rather than _Printf_

Reply:
I'm pretty confident that using Print is a no go 😉

Reply:
> I am pretty confident they do

maybe in the shell, because it both catches proper logs and stdout?

Reply:
I think logs go to stdout as well by default, it's about the format of the log message which is missing when using Print